pkg/http/sender.go

```diff
@@ -0,0 +1,638 @@
+// before any Column method.
+//
+// Symbol name cannot contain any of the following characters:
+// '\n', '\r', '?', '.', ',', ”', '"', '\\', '/', ':', ')', '(', '+',
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) Symbol(name, val string) *LineSender {
+	s.Buffer.Symbol(name, val)
+	return s
+}
+
+// Int64Column adds a 64-bit integer (long) column value to the ILP
+// message.
+//
+// Column name cannot contain any of the following characters:
+// '\n', '\r', '?', '.', ',', ”', '"', '\\', '/', ':', ')', '(', '+',
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) Int64Column(name string, val int64) *LineSender {
+	s.Buffer.Int64Column(name, val)
+	return s
+}
+
+// Long256Column adds a 256-bit unsigned integer (long256) column
+// value to the ILP message.
+//
+// Only non-negative numbers that fit into 256-bit unsigned integer are
+// supported and any other input value would lead to an error.
+//
+// Column name cannot contain any of the following characters:
+// '\n', '\r', '?', '.', ',', ”', '"', '\\', '/', ':', ')', '(', '+',
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) Long256Column(name string, val *big.Int) *LineSender {
+	s.Buffer.Long256Column(name, val)
+	return s
+}
+
+// TimestampColumn adds a timestamp column value to the ILP
+// message.
+//
+// Column name cannot contain any of the following characters:
+// '\n', '\r', '?', '.', ',', ”', '"', '\\', '/', ':', ')', '(', '+',
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) TimestampColumn(name string, ts time.Time) *LineSender {
+	s.Buffer.TimestampColumn(name, ts)
+	return s
+}
+
+// Float64Column adds a 64-bit float (double) column value to the ILP
+// message.
+//
+// Column name cannot contain any of the following characters:
+// '\n', '\r', '?', '.', ',', ”', '"', '\', '/', ':', ')', '(', '+',
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) Float64Column(name string, val float64) *LineSender {
+	s.Buffer.Float64Column(name, val)
+	return s
+}
+
+// StringColumn adds a string column value to the ILP message.
+//
+// Column name cannot contain any of the following characters:
+// '\n', '\r', '?', '.', ',', ”', '"', '\', '/', ':', ')', '(', '+',
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) StringColumn(name, val string) *LineSender {
+	s.Buffer.StringColumn(name, val)
+	return s
+}
+
+// BoolColumn adds a boolean column value to the ILP message.
+//
+// Column name cannot contain any of the following characters:
+// '\n', '\r', '?', '.', ',', ”', '"', '\', '/', ':', ')', '(', '+',
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) BoolColumn(name string, val bool) *LineSender {
+	s.Buffer.BoolColumn(name, val)
+	return s
+}
+
+// Close closes the underlying HTTP client. If no clients remain open,
+// the global http.Transport will close all idle connections.
+func (s *LineSender) Close() {
+	newCt := clientCt.Add(-1)
```

Comment:
We should check auto flush here and, if it's enabled, flush the sender here.

Also, we should have a `closed` flag in the sender, to avoid weird behavior (multiple `newCt` decrement) on double close.

Reply:
I added a context to the Close argument list to pass to `s.Flush` in the body of the function. I updated all tests and also added error checking to the `defer` calls in the examples